data/src/main/java/com/microsoft/azure/kusto/data/ConnectionStringBuilder.java

```diff
@@ -147,17 +147,31 @@ public static ConnectionStringBuilder createWithAadApplicationCredentials(String
         return createWithAadApplicationCredentials(resourceUri, applicationClientId, applicationKey, null);
     }
 
-    public static ConnectionStringBuilder createWithDeviceCodeCredentials(String resourceUri) {
+    public static ConnectionStringBuilder createWithDeviceCodeCredentials(String resourceUri, String authorityId) {
         if (StringUtils.isEmpty(resourceUri)) {
             throw new IllegalArgumentException("resourceUri cannot be null or empty");
         }
-        return new ConnectionStringBuilder(resourceUri);
+
+        ConnectionStringBuilder csb = new ConnectionStringBuilder(resourceUri);
+        csb.aadAuthorityId = authorityId;
```

Comment:
What happens if authorityId is empty?
To what will it default?

Reply:
default is microsoft.com

Reply:
"common" is better, but check with Ran that it is correct and check that it works for you correctly